mesh/amqp.c

```diff
@@ -460,125 +558,138 @@ void mesh_amqp_start(struct mesh_amqp *amqp)
 void mesh_amqp_free(struct mesh_amqp *amqp)
 {
 	if (amqp->thread_started) {
-		mesh_amqp_close(amqp);
+		mesh_amqp_stop(amqp);
 		pthread_join(amqp->thread, NULL);
 	}
 
 	l_io_destroy(amqp->io);
 	l_queue_destroy(amqp->queue, l_free);
-	l_free(amqp->url);
-	l_free(amqp->exchange);
-	l_free(amqp->routing_key);
 	l_free(amqp);
 }
 
-const char *mesh_amqp_get_url(struct mesh_amqp *amqp)
+static void *get_ret_message(struct mesh_amqp *amqp)
 {
-	return amqp->url;
+	while (1) {
```

Comment:
Ewwwwwww

Reply:
I've added a comment. Please read and say is it ok or not :)